app/controllers/halls_controller.rb

```diff
@@ -0,0 +1,49 @@
+class HallsController < ApplicationController
+before_action :set_hall, only: %i[ show edit update destroy ]
+
+  def index
+    @halls = Hall.all
+  end
+
+  def new
+    @hall = Hall.new
+    @hall_path = halls_path
+  end
+
+  def create
+    @hall = Hall.new(hall_params)
+
+    if @hall.save
+      redirect_to halls_path
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def edit
+      @hall_path = hall_path(@hall)
+  end
+
+  def update
+    if @hall.update(hall_params)
+      redirect_to @hall
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @hall.destroy
+
+    redirect_to halls_path, status: :see_other
+  end
+
+  private
```

Comment:
I would add an empty line after `private`. Right now it seems that only `set_hall` method is private even though `hall_params` is also private, empty line will make that much clearer (at least for me).